File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # using two pointer approach

        lpointer = 0
        rpointer = len(numbers)-1

        while lpointer < rpointer:
            mynum = numbers[lpointer] + numbers[rpointer]
            if mynum > target:
                rpointer -= 1
            elif mynum < target:
                lpointer += 1
            else:
                return [lpointer + 1, rpointer + 1]
            
        

            
            
            

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int lPointer = 0;
        int rPointer = numbers.length -1;
        int sum;
        int [] sumIndex = new int[2];

        while (lPointer < rPointer) {
            sum = numbers[lPointer] + numbers[rPointer];

            if (sum < target) {
                lPointer = lPointer +1;
            } else if (sum > target) {
                rPointer = rPointer - 1;                
            } else {
                sumIndex[0] = lPointer + 1;
                sumIndex[1] = rPointer + 1;
                return sumIndex;
            }
        }

        return sumIndex;
    }
}